### PR TITLE
Fix Text Alignments in Popups

### PR DESCRIPTION
**Pull request by blueskyGH**
Fixed the alignment issues in Popups.

Examples:
Before
 screenshot_2022-07-11_02-19-13
screenshot_2022-07-11_02-20-25

After
screenshot_2022-07-11_02-36-26
screenshot_2022-07-11_02-36-39

Its my first PR for actual code. Please indulge me. :) 

We still need to get rid of the "ExtraAlign" variable and only use TextWidth().
But somehow TextWidth() doesn't work properly in this case. 
Hopefully a more experienced collaborator can have a look at these lines:
https://github.com/ddnet/ddnet/pull/5615/commits/bbca48aa5b5a35e35d5174d26ff3dac5a96ff785#diff-3aebb8cc835bb888d687b51e9dcfc0cda5326d89fd1c97ff83d85c00b82ed116L1656-R1657

## Checklist
- [x] Tested the change ingame
- [x] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/client/components/menus.cpp [hunk 1]
```diff
@@ -2050,7 +2043,7 @@ int CMenus::Render()
 					{
 						const char *pError = Client()->DemoPlayer_Render(aBufOld, m_vDemos[m_DemolistSelectedIndex].m_StorageType, m_aCurrentDemoFile, m_Speed);
 						m_Speed = 4;
-						//Console()->Print(IConsole::OUTPUT_LEVEL_DEBUG, "demo_render_path", aWholePath);
+						// Console()->Print(IConsole::OUTPUT_LEVEL_DEBUG, "demo_render_path", aWholePath);
```
Comment (def-) [inline src/game/client/components/menus.cpp:2046]: Some of these whitespace changes are unexpected. I guess you are using a different clang-format version from us.

File: src/game/client/components/menus.cpp [hunk 2]
```diff
@@ -1441,43 +1442,39 @@ int CMenus::Render()
...
 				case IClient::CONNECTIVITY_DIFFERING_UDP_TCP_IP_ADDRESSES:
-					pMessage = Localize("UDP and TCP IP addresses seem to be different. Try disabling VPN, proxy or network accelerators.");
+					pExtraText = Localize("UDP and TCP IP addresses seem to be different. Try disabling VPN, proxy or network accelerators.");
 					break;
 				case IClient::CONNECTIVITY_REACHABLE:
-					pMessage = Localize("No answer from server yet.");
+					pExtraText = Localize("No answer from server yet.");
 					break;
 				}
-				if(pMessage)
```
Comment (def-) [inline src/game/client/components/menus.cpp:1467]: What is pMessage now being used for?

**Comment by def-** [comment]
There is a conflict, can you rebase your change on latest master state?

**Comment by C0D3D3V** [comment]
I guess you could rename `ExtraAlign` to `ExtraTextAlignment`. That sounds more verbose. And then define an enum like
```
	enum
	{
                TEXT_ALIGNMENT_LEFT = 0,
                TEXT_ALIGNMENT_CENTERED = 1,
	};
```
And use that instead of the magic constants 0 and -1

**Comment by C0D3D3V** [comment]
Also I wonder why you need `UseIpLabel` could you not use the ExtraText to display it centred?

**Comment by blueskyGH** [comment]
> 

There is already an if-statement that checks the TextWidth and then decides how the ExtraText should be aligned.
But it doesnt seem to work, thats why someone used ExtraAlign. As said in my pr description, maybe someone can have a look at this so that we can get rid of ExtraAlign.

File: src/game/client/components/menus.cpp (hunk 2 above)
Comment (blueskyGH) [reply src/game/client/components/menus.cpp:1467]: Nothing, it can be removed.

Comment (blueskyGH) [reply src/game/client/components/menus.cpp:1467]: I removed it

**Comment by def-** [comment]
Screenshot 2022-07-11 at 16 43 58
I'm not seeing any message now when trying to connect to a server that doesn't exist. Is this expected?

**Comment by blueskyGH** [comment]
"I'm not seeing any message now when trying to connect to a server that doesn't exist. Is this expected?"
Its the same on master, there is no message.

**Comment by def-** [comment]
Works fine for me on master, not with your change
Screenshot 2022-07-11 at 17 29 15

**Comment by def-** [comment]
My bad. I just fetch upstream again and now it also fails on master. Let me investigate.

**Pull request merged**